server/src/main/java/io/crate/planner/node/ddl/AlterTableDropCheckConstraintPlan.java

```diff
@@ -55,31 +52,18 @@ public void executeOrFail(DependencyCarrier dependencies,
                               RowConsumer consumer,
                               Row params,
                               SubQueryResults subQueryResults) {
-        dependencies.alterTableOperation()
-            .executeAlterTableAddColumn(bind(dropCheckConstraint))
-            .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
-    }
+        if (plannerContext.clusterState().getNodes().getMinNodeVersion().before(Version.V_5_1_0)) {
```

Comment:
I think we should prevent this in a generic way for all non-DML operations. Maybe hooked up while checking supported table operations https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/metadata/table/Operation.java#L119. (would require passing in the clusterState). As it's not completely straight-forward and may requires discussion, I'd suggest to do this dedicated (follow up). So either we should add an TODO comment here or remove this check.

Reply:
removed the check - let's do it dedicated

Reply:
as discussed, made it BWC with TODO
https://github.com/crate/crate/pull/12996/commits/b6ad33d61b95a68a1480c2f19160a1a980bd1d11